### ISSUE TITLE
Stronger yaml syntax checks

### ISSUE THREAD

**Issue by TobiasKadelka** [psyinfra/onyo#117]
There are a few cases, where no yaml error appears and onyo allows saving a file, that we might want to exclude in onyo, since they lead to errors.

1.)
```
f---
```
This happens when accidently adding a character before the --- of yaml. When the file contains nothing else, it seems to be excepted, but when then doing e.g. `onyo set --recursive CPU=10` it gives this error:
```
ERROR:onyo:
Can't update asset /Users/tkadelka/new-onyo/onyo_tests/onyo_tests/test_1/.onyo/temp/laptop_apple_macbookpro.1: 
'str' object does not support item assignment
```
The path is just for debugging reasons like that, the important part is the `'str' object does not support item assignment`.

2.)
```
---
RAM=10
```
The "correct" yaml syntax would be `RAM: 10`. However, the example does not lead to a yaml error, but instead it leads to the same behaviour as in 1.).

3.)
```
---
RAM=10
```
but with a validation rule specified for field RAM, leads to a different error:
```
Traceback (most recent call last):
  File "/Users/tkadelka/.venvs/onyo/bin/onyo", line 8, in <module>
    sys.exit(main())
  File "/Users/tkadelka/.venvs/onyo/lib/python3.9/site-packages/onyo/main.py", line 394, in main
    args.run(args, onyo_root)
  File "/Users/tkadelka/.venvs/onyo/lib/python3.9/site-packages/onyo/commands/edit.py", line 58, in edit
    edit_file(source, onyo_root)
  File "/Users/tkadelka/.venvs/onyo/lib/python3.9/site-packages/onyo/utils.py", line 219, in edit_file
    problem_str = validate_file(temp_file, file, onyo_root)
  File "/Users/tkadelka/.venvs/onyo/lib/python3.9/site-packages/onyo/utils.py", line 153, in validate_file
    error_str = error_str + validate_rule_for_file(file, rule, path_of_rule, original_file, onyo_root)
  File "/Users/tkadelka/.venvs/onyo/lib/python3.9/site-packages/onyo/utils.py", line 126, in validate_rule_for_file
    if not check_int(asset[value_field]):
TypeError: string indices must be integers
```
This last case is definitely an error that onyo does not catch currently, but to enable a more complete discussion about yaml syntax, I wanted to share this, too, before changing something.

4.)
```
key: 
```

`ruamel.yaml` allows this without error when it is just this one line, even though it is not valid yaml.

--------------------

So in general, I might just try to catch more errors or give "better" warnings. But especially in the first cases, one does not get a line number, just the str assignment message, and that not for the "current" change, but because of the `f---` in the beginning.

Therefore, we should discuss how much "correct yaml syntax usage" we enforce, when to print out additional error messages, and in general what onyo should do and what we want the user to do.

**Comment by TobiasKadelka** [comment]
I did not check what happens, when values have lists (e.g. `hardware: [monitor, keyboard]`), but this might be even worse, because lists are a valid use case.

So while I did not check, if they are a problem, every implemented change for the above cases should also make sure this works with lists -> Relevant issues connected to syntax checks in combination with lists are #23 #103 #114 and I think it makes sense to explore the expansion of syntax closely together with lists.
